iotdb/src/main/java/org/apache/iotdb/db/engine/memcontrol/RecordMemController.java

```diff
@@ -33,15 +32,15 @@
  */
 public class RecordMemController extends BasicMemController {
 
-  private static Logger LOGGER = LoggerFactory.getLogger(RecordMemController.class);
+  private static final Logger LOGGER = LoggerFactory.getLogger(RecordMemController.class);
 
   // the key is the reference of the memory user, while the value is its memory usage in byte
-  private Map<Object, Long> memMap;
+  private Map<Object, AtomicLong> memMap;
```

Comment:
maybe `ConcurrentHashMap ` is enough?
@jixuan1989

Reply:
ConcurrentHashMap avoids deadlock but provides no consistency.